### PR TITLE
feat(migrations): split

### PR DESCRIPTION
**Pull request by pret-a-porter**
## Changes:

1. Bunch of migrations from previously huge PR https://github.com/renovatebot/renovate/pull/12957

## Context:

1. https://github.com/renovatebot/renovate/issues/11459
2. Part of https://github.com/renovatebot/renovate/pull/12957 

## Documentation

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/config/migrations/custom/enabled-managers-migration.spec.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { EnabledManagersMigration } from './enabled-managers-migration';
+
+describe('config/migrations/custom/enabled-managers-migration', () => {
+  it('should replace yarn by nmp', () => {
```
Comment (JamieMagee) [inline lib/config/migrations/custom/enabled-managers-migration.spec.ts:4]:
```suggestion
  it('should replace yarn with npm', () => {
```

File: lib/config/migrations/custom/rebase-stale-prs-migration.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import { AbstractMigration } from '../base/abstract-migration';
+
+export class RebaseStalePrsMigration extends AbstractMigration {
+  override readonly deprecated = true;
+  readonly propertyName = 'rebaseStalePrs';
+
+  override run(value): void {
+    const rebaseConflictedPrs = this.get('rebaseConflictedPrs');
+
+    if (rebaseConflictedPrs !== false) {
```
Comment (JamieMagee) [inline lib/config/migrations/custom/rebase-stale-prs-migration.ts:11]: IMO the double negative is a little confusing

```suggestion
    if (rebaseConflictedPrs === true) {
```

or 

```suggestion
    if (rebaseConflictedPrs) {
```

File: lib/config/migrations/custom/semantic-commits-migration.spec.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
+import { SemanticCommitsMigration } from './semantic-commits-migration';
+
+describe('config/migrations/custom/semantic-commits-migration', () => {
+  it('should migrate true to "enabled"', () => {
+    expect(SemanticCommitsMigration).toMigrate(
+      {
+        semanticCommits: true,
+      } as any,
```
Comment (JamieMagee) [inline lib/config/migrations/custom/semantic-commits-migration.spec.ts:8]: Just to confirm for my sake, the `any`s are explicitly required here due to the updated types in `RenovateConfig`?

**Review by JamieMagee — DISMISSED** [review]
Just a couple of small comments from me. I love this new style of migration!

File: lib/config/migrations/custom/semantic-commits-migration.spec.ts (hunk 3 above)
Comment (pret-a-porter) [reply lib/config/migrations/custom/semantic-commits-migration.spec.ts:8]: Yes, indeed, it is string now, but previously it was boolean apparently

Comment (pret-a-porter) [reply lib/config/migrations/custom/semantic-commits-migration.spec.ts:8]: Yes, indeed, it is string now, but previously it was boolean apparently

File: lib/config/migrations/custom/rebase-stale-prs-migration.ts (hunk 2 above)
Comment (JamieMagee) [reply lib/config/migrations/custom/rebase-stale-prs-migration.ts:11]: Now I re-read this, I realise that `rebaseConflictedPrs` is `boolean | undefined` so you'd need `rebaseConflictedPrs === true || rebaseConflictedPrs === undefined`. Gotta love that trinary logic 😅

**Review by JamieMagee — APPROVED** [review]

**Pull request merged**